tests/src/test/scala/ch/renku/acceptancetests/pages/ProjectPage.scala

```diff
@@ -205,16 +218,32 @@ class ProjectPage(projectDetails: ProjectDetails, userCredentials: UserCredentia
           .getOrElse(fail("Environments -> Running title not found"))
       }
 
-      def connectToJupyterLab(implicit webDriver: WebDriver, spec: AcceptanceSpec): Unit = eventually {
+      def connectToJupyterLab(implicit webDriver: WebDriver, spec: AcceptanceSpec): Unit = {
+        connectToJupyterLab(s"a[href*='/jupyterhub/user/']")
+      }
+
+      def connectToAnonymousJupyterLab(implicit webDriver: WebDriver, spec: AcceptanceSpec): Unit = {
+        connectToJupyterLab(s"a[href*='/jupyterhub-tmp/user/']")
+      }
+
+      def connectButton(buttonSelector: String)(implicit webDriver: WebDriver): WebElement = eventually {
```

Comment:
I'm guessing this is a private method...

Reply:
I think that makes sense, but the initial implementation was also not declared private, and none of the other {foo}Button methods are private either, so I left it with default visibility. What do you think?

Reply:
It's not the end of the world to leave it public but if I think it was left public by accident which I suppose it shouldn't.

Reply:
I suggest leaving it public, since all the other similar methods are public, but I think it might make sense to go through and convert all to private.